Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -768,7 +769,40 @@ extension Tensor: Differentiable & EuclideanDifferentiable where Scalar: TensorF
         case .XLA:
           return xlaTensor.device
         case .TF_EAGER:
-          return Device.defaultTFEager
+          var kind: Device.Kind = .CPU
+          var ordinal = 0
+          let status = _ExecutionContext.global.status
+
+          // Find out what the underlying libraries think the default is.
+          if let cString = TFE_TensorHandleDeviceName(handle._cTensorHandle, status) {
+            checkOk(status)
+            let tfDeviceName = String(cString: cString)
+
+            // Parse type and ordinal from a string with the expected syntax:
+            //   /job:localhost/replica:0/task:0/device:CPU:0
+            let pattern = ".+device:(.+):(\\d+)$"
```

Comment:
Maybe break this String -> Device out as a separate function?
Also, I'm concerned that the string parsing will be expensive. This function is called a lot. (whenever there is a scalar constant or anything like that). I think it would be best to see if you can add your own TFE_TensorHandleDevice_Type and TFE_TensorHandleDevice_Id. Some benchmarking results might work instead.